src/apifetch.js

```diff
@@ -69,163 +75,237 @@ var executeApiFetch = function (
-          '&hierarchicalFacets=' +
-          encodeURIComponent(JSON.stringify(settings.hierarchicalFacetSetting));
-      }
+    // Range facets
+    if (settings.rangeFacets) {
+      queryParamsString =
+        queryParamsString +
+        '&rangeFacets=' +
+        encodeURIComponent(JSON.stringify(settings.rangeFacets));
+    }
 
-      // Stats fields
-      if (settings.statsFields) {
-        for (var i = 0; i < settings.statsFields.length; i++) {
-          qs = qs + '&fieldStat=' + settings.statsFields[i];
-        }
-      }
+    // Hierarchical facets
+    if (settings.hierarchicalFacetSetting) {
+      queryParamsString =
+        queryParamsString +
+        '&hierarchicalFacets=' +
+        encodeURIComponent(JSON.stringify(settings.hierarchicalFacetSetting));
+    }
 
-      // Personalization events
-      if (settings.personalizationEvents && Array.isArray(settings.personalizationEvents)) {
-        for (let i = 0; i < settings.personalizationEvents.length; i++) {
-          var obj = settings.personalizationEvents[i];
-          var key = Object.keys(obj);
-          qs = qs + '&personalizationEvent=' + encodeURIComponent(key + '=' + obj[key]);
-        }
+    // Stats fields
+    if (settings.statsFields) {
+      for (var i = 0; i < settings.statsFields.length; i++) {
+        queryParamsString = queryParamsString + '&fieldStat=' + settings.statsFields[i];
       }
+    }
 
-      // Filter object
-      if (customFilterObject) {
-        qs = qs + '&filter=' + encodeURIComponent(JSON.stringify(customFilterObject));
-      } else if (settings.filterObject) {
-        qs = qs + '&filter=' + encodeURIComponent(JSON.stringify(settings.filterObject));
+    // Personalization events
+    if (settings.personalizationEvents && Array.isArray(settings.personalizationEvents)) {
+      for (let i = 0; i < settings.personalizationEvents.length; i++) {
+        var obj = settings.personalizationEvents[i];
+        var key = Object.keys(obj);
+        queryParamsString =
+          queryParamsString + '&personalizationEvent=' + encodeURIComponent(key + '=' + obj[key]);
       }
     }
-    api = 'https://' + apiHostname + '/v1/' + apiPath + '/' + sitekey + '?term=' + kw + qs;
+
+    // Filter object
+    if (customFilterObject) {
+      queryParamsString =
+        queryParamsString + '&filter=' + encodeURIComponent(JSON.stringify(customFilterObject));
+    } else if (settings.filterObject) {
+      queryParamsString =
+        queryParamsString + '&filter=' + encodeURIComponent(JSON.stringify(settings.filterObject));
+    }
+
+    apiEndpoint =
+      'https://' +
+      apiHostname +
+      '/v1/' +
+      apiPath +
+      '/' +
+      sitekey +
+      '?term=' +
+      keyword +
+      queryParamsString;
+  }
+
+  // Conversational Search
+  else if (type === 'conversational-search') {
+    // TODO use apiHostname instead of hardcoded URL
+    apiInstance
+      .post(`https://api.addsearch.com/v2/indices/${sitekey}/conversations`, {
```

Comment:
Hardcoded URL can be replaced
It's not critical feature, but maybe we will also test conversational search in staging env at some point.

Reply:
Yeah, for now that there is only one endpoint I think it needs to stay like this.